src/test/java/uk/gov/ons/ctp/response/casesvc/endpoint/CaseGroupEndpointUnitTest.java

```diff
@@ -129,20 +132,27 @@ public void findCaseGroupById() throws Exception {
    */
   @Test
   public void findCaseGroupByIdNotFound() throws Exception {
-    ResultActions actions =
-        mockMvc.perform(getJson(String.format("/casegroups/%s", NON_EXISTENT_CASE_GROUP_UUID)));
-
-    actions.andExpect(status().isNotFound());
-    actions.andExpect(handler().handlerType(CaseGroupEndpoint.class));
-    actions.andExpect(handler().methodName("findCaseGroupById"));
-    actions.andExpect(jsonPath("$.error.code", is(CTPException.Fault.RESOURCE_NOT_FOUND.name())));
-    actions.andExpect(
-        jsonPath(
-            "$.error.message",
-            is(
-                String.format(
-                    "CaseGroup not found for casegroup id %s", NON_EXISTENT_CASE_GROUP_UUID))));
-    actions.andExpect(jsonPath("$.error.timestamp", isA(String.class)));
+    Exception exception =
+        assertThrows(
+            Exception.class,
+            () ->
+                mockMvc
+                    .perform(getJson(String.format("/casegroups/%s", NON_EXISTENT_CASE_GROUP_UUID)))
+                    .andExpect(status().isNotFound())
+                    .andExpect(handler().handlerType(CaseGroupEndpoint.class))
+                    .andExpect(handler().methodName("findCaseGroupById"))
+                    .andExpect(jsonPath("$.error.code", is(RESOURCE_NOT_FOUND.name())))
+                    .andExpect(
+                        jsonPath(
+                            "$.error.message",
+                            is(
+                                String.format(
+                                    "CaseGroup not found for casegroup id %s",
+                                    NON_EXISTENT_CASE_GROUP_UUID))))
+                    .andExpect(jsonPath("$.error.timestamp", isA(String.class))));
+
+    assertTrue(exception.getMessage().contains(NON_EXISTENT_CASE_GROUP_UUID));
+    assertTrue(exception.getMessage().contains("CaseGroup not found for casegroup id"));
```

Comment:
The test framework under Spring Boot 3 now causes a test error when throwing an exception during the test. We now need to use `assertThrows` to bury that and be able to use it during the test.